src/cache.c

```diff
@@ -82,6 +82,13 @@ static struct cmusfm_cache_record *get_cache_record(const scrobbler_trackinfo_t
 
 	/* enlarge allocated memory for string data payload */
 	record = (struct cmusfm_cache_record *)realloc(record, get_cache_record_size(record));
+
+	if (record == NULL) {
+		perror("ERROR: error enlarge allocated memory");
+		free(record);
+		exit(EXIT_FAILURE);
```

Comment:
Yes, technically that's a bad codding. However, this function does not check for `malloc()` failure either. If you would like to "fix" it (side note: on most Linux distros malloc/realloc does not fail due to memory overcommit) you should check for malloc failure as well. Also, I don't like calling `exit()`. This function should return `NULL` in case of error. Then, the caller should check for NULL when calling `get_cache_record()` and act accordingly.

Reply:
Yes, you are absolutely right, I read [Memory Management](https://www.kernel.org/doc/html/latest/admin-guide/mm/index.html) today (sorry that I did not read it before the PR) and came to the conclusion that a failure to allocate memory in linux is an unlikely event, and most likely it will not happen, but if it does, then this is a completely different problem level and there is no longer up to cmusfm. It makes sense to correct only with a tendency to perfectionism. I'm sorry for wasting your time. I think to close this PR.